Test/ParallelPrimitives/RadixSort.cpp

```diff
@@ -1,58 +1,121 @@
+#include <Test/OrochiUtils.h>
 #include <Test/ParallelPrimitives/RadixSort.h>
 #include <Test/ParallelPrimitives/RadixSortConfigs.h>
-#include <Test/OrochiUtils.h>
+#include <numeric>
+
+namespace
+{
+
+/// @brief Exclusive scan algorithm on CPU for testing.
+/// It copies the count result from the Device to Host before computation, and then copies the offsets back from Host to Device afterward.
+/// @param countsGpu The count result in GPU memory. Otuput: The offset.
+/// @param offsetsGpu The offsets.
+/// @param nWGsToExecute Number of WGs to execute
+void exclusiveScanCpu( int* countsGpu, int* offsetsGpu, const int nWGsToExecute ) noexcept
+{
+	std::vector<int> counts( Oro::BIN_SIZE * nWGsToExecute );
+	OrochiUtils::copyDtoH( counts.data(), countsGpu, Oro::BIN_SIZE * nWGsToExecute );
+	OrochiUtils::waitForCompletion();
+
+	constexpr auto ENABLE_PRINT{ false };
+
+	if constexpr( ENABLE_PRINT )
+	{
+		for( int j = 0; j < nWGsToExecute; j++ )
+		{
+			for( int i = 0; i < Oro::BIN_SIZE; i++ )
+			{
+				printf( "%d, ", counts[j * Oro::BIN_SIZE + i] );
+			}
+			printf( "\n" );
+		}
+	}
+
+	std::vector<int> offsets( Oro::BIN_SIZE * nWGsToExecute );
+	std::exclusive_scan( std::cbegin( counts ), std::cend( counts ), std::begin( offsets ), 0 );
+
+	OrochiUtils::copyHtoD( offsetsGpu, offsets.data(), Oro::BIN_SIZE * nWGsToExecute );
+	OrochiUtils::waitForCompletion();
+}
+
+} // namespace
 
 namespace Oro
 {
 
 struct RadixSortImpl
 {
-	static
-	void printKernelInfo( oroFunction func ) 
-	{ 
+	static void printKernelInfo( oroFunction func )
```

Comment:
(Some of the format is changed automatically by Visual Studio using the clangformat when the corresponding file is closed.)